ex16/part3/src/index.js

```diff
@@ -0,0 +1,62 @@
+import React, {useState} from "react";
+import ReactDOM from "react-dom";
+import Form1 from "./form1";
+import Form2 from "./form2";
+import Form3 from "./form3";
+
+const App = () => {
+    const [step, setStep] = useState(0);
+    const [formData, setFormData] = useState({name: "", pasword: "", cuntry: "", city: ""})
+
+    function nextStep(e) {
+        if (step < 2) {
+            setStep(s => s + 1)
+        }
+    }
+
+    function prevStep(e) {
+        if (step > 0) {
+            setStep(s => s - 1)
+        }
+    }
+
+    function handleChange(e) {
+        const {name, value} = e.target
+        setFormData(data => ({
+            ...formData,
+            [name]: value
+        }));
+    }
+
+    function getForm(){
+        switch (step) {
+            case 0: return <Form1 data={formData} seter={handleChange}/>
```

Comment:
It is a good solution, however it is better not to put big chunks of the JSX outside the return function.
It would be more clear this way:
```js
{step === 1 && <Form1 data={formData} seter={handleChange}/>}
```